### ISSUE TITLE
How to serve both static and dynamic content via Nginx from php-fpm container?

### ISSUE THREAD

**Issue by zigi05** [bitnami/bitnami-docker-nginx#138]
My app files are located in `phpfpm` container and I need to serve them through `nginx`. However, when I use _**Reverse proxy to other containers**_ as instructed in the _readme_:
```
server {
    listen 0.0.0.0:8080;
    server_name myapp.test;
    access_log /opt/bitnami/nginx/logs/myapp_access.log;
    error_log /opt/bitnami/nginx/logs/myapp_error.log;

    location / {
        proxy_set_header X-Real-IP $remote_addr;
        proxy_set_header HOST $http_host;
        proxy_set_header X-NginX-Proxy true;

        proxy_pass http://phpfpm:900;
        proxy_redirect off;
    }
}
```
I get _**502 Bad Gateway**_ error with the following error log record:

> 1 recv() failed (104: Connection reset by peer) while reading response header from upstream, client: 172.18.0.1, server: myapp.test, request: "GET / HTTP/1.1", upstream: "http://172.18.0.2:9000/", host: "myapp.test"

I guess it's because `phpfpm` container is not a HTTP proxied server.

So, alternatively, I try using `fastcgi_pass` like so:
```
server {
    listen 0.0.0.0:8080;
    server_name myapp.test;

    root /app;

    location / {
        try_files $uri $uri/index.php;
    }

    location ~ \.php$ {
        fastcgi_pass phpfpm:9000;
        fastcgi_index index.php;
        include fastcgi.conf;
    }
}
```
This serves `*.php` files as expected, but doesn't serve other files, namely static content.

How do I make`nginx` serve both dynamic and static files from my `phpfpm` container?

Here's my _docker-compose.yml_:
```yaml
version: "3.7"
services:
  phpfpm:
    image: "bitnami/php-fpm:7.3"
    volumes:
      - ./site:/app
    ports:
      - "9000:9000"

  nginx:
    image: "bitnami/nginx:1.17"
    volumes:
      - ./_bitnami/nginx/app.conf:/opt/bitnami/nginx/conf/server_blocks/nginx.conf
    ports:
      - "80:8080"
```

**Comment by juan131** [comment]
Hi @zigi05 

I guess you'll have to put your website (both static and PHP files) under a volume. Then, mount this volume both on the NGINX and the PHP-FPM container. This way you ensure both containers have access to the files.

Once you accomplish that, it's a matter of configuring your server blocks correctly, so NGINX redirect PHP requests to the PHP-FPM container, while it serves the rest of the files directly from the NGINX container.

Does it make sense?

**Comment by 1212087** [comment]
> Hi @zigi05
> 
> I guess you'll have to put your website (both static and PHP files) under a volume. Then, mount this volume both on the NGINX and the PHP-FPM container. This way you ensure both containers have access to the files.
> 
> Once you accomplish that, it's a matter of configuring your server blocks correctly, so NGINX redirect PHP requests to the PHP-FPM container, while it serves the rest of the files directly from the NGINX container.
> 
> Does it make sense?

Hi @juan131 , 
I did as your solution before, but there is one problem here: when you re-create your containers from a newer version of the image (image also contain some public files such as JS, CSS bundle...), these new files will be replaced by the old ones from the mounted volume. I could delete the volume before re-creating the containers to make sure that all new files from phpfpm -> volume -> nginx, but are there any better solutions?

**Comment by juan131** [comment]
Let's continue the conversation in the new issue you just created and keep this one closed @1212087
